File: computercraft/execute_reversed_moves.py
from typing import Sequence, List

from .execute_moves import execute_moves
from .constants import Constants, Mock

turtle = Mock.Turtle()


def execute_reversed_moves(moves: Sequence[str]) -> None:
    """
    This function turns the turtle 180 degrees before and after exevuting the reversed move sequence,
    and therefore does not reverse forward and back moves. This approach is in the interest of time efficiency
    """

    turtle.turnRight()
    turtle.turnRight()

    execute_moves([Constants.reversed_moves[move] for move in reversed(moves)])

    turtle.turnRight()
    turtle.turnRight()

File: computercraft/execute_reversed_moves.lua
local execute_moves = require("execute_moves")
local constants = require("constants")


function execute_reversed_moves(moves)
    --[[
    This function turns the turtle 180 degrees before and after exevuting the reversed move sequence,
    and therefore does not reverse forward and back moves. This approach is in the interest of time efficiency
    --]]

    local reversed_moves = {}
    for index, move in ipairs(moves) do
        reverse_index = #moves - (index-1)

        reversed_moves[reverse_index] = constants.reversed_moves[move]
    end

    turtle.turnRight()
    turtle.turnRight()

    execute_moves(reversed_moves)

    turtle.turnRight()
    turtle.turnRight()
end


return execute_reversed_moves
